can/io/player.py

```diff
@@ -45,45 +46,51 @@ class LogReader(BaseIOHandler):
     """
 
     @staticmethod
-    def __new__(cls, filename, *args, **kwargs):
+    def __new__(cls, filename: "can.typechecking.PathLike", *args, **kwargs):
         """
-        :param str filename: the filename/path the file to read from
+        :param filename: the filename/path the file to read from
```

Comment:
Nit: Unrelated to your changes, but I think we're missing a word here in the docstring?

```suggestion
        :param filename: the filename/path of the file to read from
```